### PR TITLE
chore: update docker-compose to use healthcheck instead of wait-for-it script

### PR DESCRIPTION
**Pull request by philippechevieux**
## Checklist

Definition Of Review

-   [x] Own code review done
-   [x] Write message in teams channel

Definition Of Mergeable

-   [x] 2 approves
-   [x] 1 functional review - US has been tested
-   [x] Every comment is handled - blocking ones have been resolved by reviewer
-   [ ] Design OK
-   [ ] Can be tested by POs
-   [x] PR was introduced during daily meeting

### CODE REVIEW

File: docker/docker-compose.prod.yml [hunk 1]
```diff
@@ -195,13 +196,10 @@ services:
...
-      - message-broker
-      - elasticsearch
-    command: [
-        sh,
-        -c,
-        "/app/scripts/wait-for-it.sh message-broker:5672 -t 1000 -- yarn run start",
-      ]
+      message-broker:
+        condition: service_healthy
+      elasticsearch: {}
```
Comment (P0ppoff) [inline docker/docker-compose.prod.yml:201]: **question:** est ce que l'on peut pas trouver un healthcheck quand même ?

**Review by P0ppoff — APPROVED** [review]

File: docker/docker-compose.prod.yml (hunk 1 above)
Comment (P0ppoff) [reply docker/docker-compose.prod.yml:201]: pour example : https://github.com/elastic/elasticsearch/blob/8.10/docs/reference/setup/install/docker/docker-compose.yml

Comment (philippechevieux) [reply docker/docker-compose.prod.yml:201]: @P0ppoff Ca marche plutôt bien comme ceci:
Capture d’écran 2024-08-12 à 16 35 20
Capture d’écran 2024-08-12 à 16 38 11

Comment (P0ppoff) [reply docker/docker-compose.prod.yml]: pour curl tu as l'option `--fail ` pour échouer si code retour non 2XX; ça évite le || exit 1

Comment (philippechevieux) [reply docker/docker-compose.prod.yml]: @P0ppoff Bonne idée, j'ai fais le changement 👍

**Review by evoiron — APPROVED** [review]

**Pull request merged**